edweiss-app/app/(app)/courses/[id]/deck/[deckId]/shareDeckCard.tsx

```diff
@@ -0,0 +1,97 @@
+import { ApplicationRoute } from '@/constants/Component';
+import React from 'react';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TView from '@/components/core/containers/TView';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import { callFunction, CollectionOf } from '@/config/firebase';
+import { useAuth } from '@/contexts/auth';
+import { useUser } from '@/contexts/user';
+import { useDynamicDocs } from '@/hooks/firebase/firestore';
+import { useStringParameters } from '@/hooks/routeParameters';
+import Memento from '@/model/memento';
+import { AppUser, UserID } from '@/model/users';
+import { router } from 'expo-router';
+import { useState } from 'react';
+
+const ShareScreen: ApplicationRoute = () => {
+    const { id: courseId, deckId, type, indices_of_cards_to_share } = useStringParameters();
+    const { uid } = useAuth();
+    const { user } = useUser();
+    const [loading, setIsLoading] = useState(false);
+
+    const current_user_type = user.type
+
+    const users = useDynamicDocs(CollectionOf<AppUser>('users'));
+    if (!users) return null;
```

Comment:
I have seen you use if(variable == undefined) and now if (!users) when I was doing the overview of you index function, I would suggest choosing one of the two methods for consistency but it is up to you since it is a detail

Reply:
Yup :D